lib/WatchIgnorePlugin.js

```diff
@@ -49,5 +49,24 @@ class IgnoringWatchFileSystem {
 
 			callback(err, filesModified, dirsModified, missingModified, fileTimestamps, dirTimestamps);
 		}, callbackUndelayed);
+
+		return {
+			close: () => watcher.close(),
+			pause: () => watcher.pause(),
+			getContextTimestamps: () => {
+				const dirTimestamps = watcher.getContextTimestamps();
+				ignoredDirs.forEach(path => {
+					dirTimestamps[path] = 1;
```

Comment:
Using a `Set` would prevent deopt

Reply:
Good idea to refactor this. -> separate PR.

Reply:
Created https://github.com/webpack/webpack/issues/6234 for the refactor.

Reply:
Note: It should be a `Map`